addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/automation/actions/RunScriptAction.java

```diff
@@ -131,7 +185,12 @@ public void runJob(String jobName, AutomationEnvironment env, AutomationProgress
                                 parameters.getName()));
                 return;
             }
-            extScript.invokeScript(script);
+            if (parameters.getType().equals(ExtensionScript.TYPE_TARGETED)) {
+                URI uri = new URI(parameters.getTarget(), true);
+                extScript.invokeTargetedScript(script, new HttpMessage(uri));
```

Comment:
I'd expect this to use a message from the Sites tree, which has the proper headers/bodies.

Reply:
For Sites tree, the Target URL has to be already present. So, are we expecting that no new (& unknown) URL can be passed into Run action directly?

Reply:
Right, the user can use the `requestor` job (or any other job that causes messages to be accessed) beforehand.